src/pages/resultsView/ResultsViewPage.tsx

```diff
@@ -61,24 +73,27 @@ export default class ResultsViewPage extends React.Component<IResultsViewPagePro
 
     public exposeComponentRenderersToParentScript(){
 
-        exposeComponentRenderer('renderMutationsTab', (props:{genes:string[], studyId:string, samples:string[]|string})=>{
-            const resultsViewPageStore = new ResultsViewPageStore();
-            resultsViewPageStore.hugoGeneSymbols = props.genes;
-            resultsViewPageStore.studyId = props.studyId;
-            if (typeof props.samples === "string") {
-                resultsViewPageStore.sampleListId = props.samples;
-            } else {
-                resultsViewPageStore.sampleList = props.samples;
-            }
+        exposeComponentRenderer('renderMutationsTab',
+            (props:MutationsTabInitProps)=>{
+                resultsViewPageStore.hugoGeneSymbols = props.genes;
+                if (props.studyToSampleMap) {
+                    let _studyToSampleIds:{[studyId:string]:{[sampleId:string]:boolean}} = {};
+                    _.each(props.studyToSampleMap, (sampleIds:string[], studyId:string)=>{
+                        _studyToSampleIds[studyId] = stringListToSet(sampleIds);
```

Comment:
also, it's confusing that we keep list of string sampleIds in study buckets and then have another flat sampleId collection.